pkg/controller/s3/bucket/lifecycleConfig.go

```diff
@@ -129,12 +130,17 @@ func GenerateRules(in *v1beta1.BucketLifecycleConfiguration) []awss3.LifecycleRu
 		if local.Filter != nil {
 			rule.Filter = &awss3.LifecycleRuleFilter{
 				Prefix: local.Filter.Prefix,
-				Tag:    testing.CopyTag(local.Filter.Tag),
+			}
+			if local.Filter.Tag != nil {
+				rule.Filter.Tag = s3.CopyTag(local.Filter.Tag)
 			}
 			if local.Filter.And != nil {
 				rule.Filter.And = &awss3.LifecycleRuleAndOperator{
 					Prefix: local.Filter.And.Prefix,
-					Tags:   testing.CopyTags(local.Filter.And.Tags),
+				}
+				if local.Filter.And.Tags != nil {
+					rule.Filter.And.Tags = s3.CopyTags(local.Filter.And.Tags)
```

Comment:
```suggestion
					rule.Filter.And.Tags = &s3.Tag{Key:   aws.String(tag.Key),Value: aws.String(tag.Value)}
```

That was what I meant by removing `CopyTag` function.